tests/pages/regions/modal.py

```diff
@@ -0,0 +1,26 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as expected
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+from ..page import PageRegion
+
+
+class Modal(PageRegion):
+
+    _root_locator = (By.ID, 'modal')
+    _close_locator = (By.ID, 'modal-close')
+
+    def close(self):
+        modal = self.selenium.find_element(*self._root_locator)
+        self.root.find_element(*self._close_locator).click()
+        Wait(self.selenium, self.timeout).until(expected.staleness_of(modal))
+
+    @property
+    def is_displayed(self):
+        Wait(self.selenium, self.timeout).until(
+            expected.visibility_of_element_located(self._root_locator))
```

Comment:
This wait will cause a timeout to elapse if we want to check that the modal is _not_ displayed. I would suggest having this property return faster, and using the return value as the condition of a wait whenever we expect the modal to become visible. For example:

```python
def play_video(self):
    self.selenium.find_element(*self._video_play_locator).click()
    Wait(self.selenium, self.timeout).until(lambda s: Modal(s).is_displayed)
```